pepsico/comparison.py

```
@@ -1,14 +1,136 @@
 import xarray as xr
+import os
+import re
+
+
+ds_zarr = '/data/remic/mydatafiles/zarr/test/hurs'
+ds_nc_dir = '/Data/data24/ISIMIP3b/InputData/climate/atmosphere/bias-adjusted/global/daily/historical/MPI-ESM1-2-HR/'
+
+#only using NetCDF files that contain 'hurs' variable
+def contains_hurs(filename):
+    return 'hurs' in filename
+
+#extracting years from nc file
+def extract_years(filename):
+    match = re.search(r'(\d{4})_(\d{4})\.nc$', filename)
+    if match:
+        start_year = int(match.group(1))
+        end_year = int(match.group(2))
+        return start_year, end_year
+    else:
+        raise ValueError("Unable to find years")
+
+
+#function that compares nc files(hurs) vs zarred file
+def compare_datasets(nc_file_path, ds_zarr):
+    #returns true if datasets are identical, false otherwise
+
+    #open datasets
+    zarr_dSet = xr.open_zarr(ds_zarr)
+    nc_dSet = xr.open_dataset(nc_file_path)[['hurs']]
+
+
+    #extracting start and end years from nc file
+    start_year, end_year = extract_years(os.path.basename(nc_file_path))
+    time_start = f'{start_year}-01-01'
+    time_end = f'{end_year}-12-31'
+
+    #check if datasets are empty
+    if len(nc_dSet) == 0 or len(zarr_dSet) == 0:
+        print("Datasets are empty")
+        return False
+
     
-# defining the path to file  
-ds_zarr = '../data/remic/mydatafiles/zarr/test/'
+    #mapping from zar to nc dimensions (X,Y,T) vs (lon,lat,time)
+    coord_map = {'lon': 'X', 'lat': 'Y', 'time': 'T'}
+
+    #xararay rename function to reset nc labels to zarr
+    nc_dSet = nc_dSet.rename({orig_coord: new_coord for orig_coord, new_coord in coord_map.items() if orig_coord in nc_dSet.coords})
+    
+    nc_dSet_sliced = nc_dSet.sel(T=slice(time_start, time_end))
```

Why? We got time_start and time_end from the filename. Do you think there might be time points in the file that are outside the range indicated by the filename? I doubt it, but if by some chance that does happen, it would be better to flag that so we can look into it, rather than silently discarding the extra data.

---

Right, that slicing is unnecessary, we can later test the unlikely case where the nc filename doesn't agree with the nc content.